### DISCUSSION TITLE
Cannot get the sample to work (Docker Desktop 4.35.1, Mac Sequoia 15.1)

### DISCUSSION THREAD

**Discussion by jimmcslim** [almeidapaulopt/tsdproxy#43] [Q&A]
Trying to get this going, following the main sample with a sample `nginx` container.

The `tsdproxy` and `nginx` containers come up, but am unable to curl the `nginx` container.

Logs from `tsdproxy` are:

````
❯ docker logs tsdproxy-tailscale-docker-proxy-1
Initializing server
Version
Setting up logger
7:58AM INF Log Settings Log level=info
7:58AM INF Starting server Version=
7:58AM INF Setting up proxy for existing containers
7:58AM INF Initializing WebServer
7:58AM INF Health check set to ready
7:58AM INF request host=127.0.0.1:8080 method=GET status=200 url=/health/ready/
7:58AM INF setting up proxy for container containerID=<hidden>
7:58AM INF tsnet running state path /data/sample-nginx/tailscaled.state
7:58AM INF tsnet running state path /data/sample-nginx/tailscaled.state
7:58AM INF tsnet running state path /data/sample-nginx/tailscaled.state
7:58AM INF tsnet running state path /data/sample-nginx/tailscaled.state
7:59AM INF request host=127.0.0.1:8080 method=GET status=200 url=/health/ready/
2024/11/15 07:59:33 http: proxy error: context canceled
7:59AM ERR error host=sample-nginx.<hidden>.ts.net method=GET status=502 url=/
7:59AM INF request host=127.0.0.1:8080 method=GET status=200 url=/health/ready/
8:00AM INF request host=127.0.0.1:8080 method=GET status=200 url=/health/ready/
````

Compose file has `TSDPROXY_HOSTNAME=localhost` - but also didn't work with trying a fixed IP either.

Running `curl https://sample-nginx.<hidden>.ts.net` I eventually timeout.

I do see `sample-nginx` in my Machines list in the Tailscale admin console.

I can successfully `curl localhost:8181` and get the response from `sample-nginx`

Update #1: if I let `curl` run and timeout, then I observe in the logs for the proxy...

````
2024/11/16 06:46:16 http: proxy error: dial tcp 172.17.0.2:8181: i/o timeout
6:46AM ERR error host=sample-nginx.<hidden>.ts.net method=GET status=502 url=/
````

Not sure where the 172.17.0.2 ip address is coming from. Shouldn't this be the IP/hostname from `TSDPROXY_HOSTNAME` that I am running Docker on?

Update #2: ok, if I start the `sample-nginx` container with `--network=tsdproxy_default` (`tsdproxy` being the name of the dir that my compose file is in - so influenced from that?) then everything works. So unless there's some different behaviour between Mac and Linux... probably a gap in my understanding of Docker networks.

**Comment by yashwantrkamath** [comment]
I am in your same positon until Update #1 

can you please help me understand what you did you did with #2 ?

**Comment by almeidapaulopt** [comment]
I'm sorry I only read this thread today.

TSDProxy try to detect a docker network to use, if for any case it can't detect it right, you can use https://almeidapaulopt.github.io/tsdproxy/docs/docker/#tsdproxyautodetect .

There are a lot of people using it with docker networks, without a network shared with tsdproxy.

Maybe in the future we should have more time analysis with docker networks to get better detection.

**Reply by jimmcslim** [reply]
I'm still not able to get this work across multiple configurations... running locally on my Mac, running on an Ubuntu VM. Is it possible there is some setting in my Tailscale configuration that is preventing anything from happening?

**Reply by jimmcslim** [reply]
I revisted the Tailscale sidecar container approach, and that seems to work fine for me. It is much more fiddly though! Should my tsdproxy compose file include `net_admin` and `sys_module` capabilities for the `tsdproxy` instance - and a link to the `/dev/net/tun` device perhaps?
